File: android/app/src/main/java/com/harshbshah/fer/data/model/MuscleGroup.kt
package com.harshbshah.fer.data.model

import androidx.compose.ui.graphics.Color

/**
 * Enum constants are deliberately lowercase (not SCREAMING_CASE) so that
 * Firestore's default `.name`-based enum serialization produces the exact
 * same raw values ("chest", "back", ...) as Swift's `Codable` on iOS —
 * required for the two platforms to read/write the same documents.
 */
enum class MuscleGroup {
    chest, back, shoulders, biceps, triceps, forearms,
    abs, quads, hamstrings, glutes, calves, cardio, fullBody;

    val displayName: String
        get() = when (this) {
            chest -> "Chest"
            back -> "Back"
            shoulders -> "Shoulders"
            biceps -> "Biceps"
            triceps -> "Triceps"
            forearms -> "Forearms"
            abs -> "Abs"
            quads -> "Quads"
            hamstrings -> "Hamstrings"
            glutes -> "Glutes"
            calves -> "Calves"
            cardio -> "Cardio"
            fullBody -> "Full Body"
        }

    val accentColor: Color
        get() = when (this) {
            chest -> Color(0xFFE53935)
            back -> Color(0xFF1E88E5)
            shoulders -> Color(0xFFFB8C00)
            biceps, triceps, forearms -> Color(0xFF8E24AA)
            abs -> Color(0xFFFDD835)
            quads, hamstrings, glutes -> Color(0xFF43A047)
            calves -> Color(0xFF00BFA5)
            cardio -> Color(0xFFD81B60)
            fullBody -> Color(0xFF00897B)
        }
}

enum class Equipment {
    barbell, dumbbell, machine, cable, bodyweight, kettlebell, band, other;

    val displayName: String
        get() = when (this) {
            barbell -> "Barbell"
            dumbbell -> "Dumbbell"
            machine -> "Machine"
            cable -> "Cable"
            bodyweight -> "Bodyweight"
            kettlebell -> "Kettlebell"
            band -> "Band"
            other -> "Other"
        }
}

File: Fer/Models/MuscleGroup.swift
//
//  MuscleGroup.swift
//  Fer
//
//  Core taxonomy used to categorize exercises.
//

import SwiftUI

enum MuscleGroup: String, Codable, CaseIterable, Identifiable {
    case chest, back, shoulders, biceps, triceps, forearms
    case abs, quads, hamstrings, glutes, calves, cardio, fullBody

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .chest: return "Chest"
        case .back: return "Back"
        case .shoulders: return "Shoulders"
        case .biceps: return "Biceps"
        case .triceps: return "Triceps"
        case .forearms: return "Forearms"
        case .abs: return "Abs"
        case .quads: return "Quads"
        case .hamstrings: return "Hamstrings"
        case .glutes: return "Glutes"
        case .calves: return "Calves"
        case .cardio: return "Cardio"
        case .fullBody: return "Full Body"
        }
    }

    var icon: String {
        switch self {
        case .chest: return "figure.strengthtraining.traditional"
        case .back: return "figure.rower"
        case .shoulders: return "figure.arms.open"
        case .biceps, .triceps, .forearms: return "dumbbell.fill"
        case .abs: return "figure.core.training"
        case .quads, .hamstrings, .glutes: return "figure.squat"
        case .calves: return "figure.step.training"
        case .cardio: return "heart.fill"
        case .fullBody: return "figure.mixed.cardio"
        }
    }

    var accentColor: Color {
        switch self {
        case .chest: return .red
        case .back: return .blue
        case .shoulders: return .orange
        case .biceps, .triceps, .forearms: return .purple
        case .abs: return .yellow
        case .quads, .hamstrings, .glutes: return .green
        case .calves: return .mint
        case .cardio: return .pink
        case .fullBody: return .teal
        }
    }
}

enum Equipment: String, Codable, CaseIterable, Identifiable {
    case barbell, dumbbell, machine, cable, bodyweight, kettlebell, band, other

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .barbell: return "Barbell"
        case .dumbbell: return "Dumbbell"
        case .machine: return "Machine"
        case .cable: return "Cable"
        case .bodyweight: return "Bodyweight"
        case .kettlebell: return "Kettlebell"
        case .band: return "Band"
        case .other: return "Other"
        }
    }
}
